src/BDCS/DB.hs

```diff
@@ -229,6 +229,12 @@ firstKeyResult :: Monad m => m [Value a] -> m (Maybe a)
 firstKeyResult query =
     listToMaybe . map unValue <$> query
 
+-- | Run an SQL query, returning the first value from the result list.
+-- Use this when you want a single index out of the database and it is guaranteed not to be empty.
+firstListResult :: Monad m => m [Value a] -> m a
+firstListResult query =
+    head . map unValue <$> query
+
 -- | Like 'maybe', but for keys.  If the key is nothing, return the default value.  Otherwise,
 -- run the function on the key and return that value.
 maybeKey :: MonadIO m =>
```

Comment:
To continue my war on partial functions, would it be possible to write firstListResult like so:
```
firstListResult :: (MonadError String m, MonadIO m) => m [Value a] -> m a
firstListResult query = (map unValue <$> query) >>= \case
    [] -> throwError "No result found"
    v:_ -> return v
```

Or something like that?  I'm not sure if the extra constraints there make a mess of callers or not.